packages/core/src/codewhisperer/models/errors.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { ToolkitError } from '../../shared/errors'
+import {
+    DefaultCodeScanErrorMessage,
+    FileSizeExceededErrorMessage,
+    NoWorkspaceFoundErrorMessage,
+    ProjectSizeExceededErrorMessage,
+    InvalidSourceFilesErrorMessage,
+    UploadArtifactToS3ErrorMessage,
+} from './constants'
+
+export enum ErrorCodes {
+    FileSizeExceeded = 'FileSizeExceeded',
+    ProjectSizeExceeded = 'ProjectSizeExceeded',
+    NoWorkspaceFound = 'NoWorkspaceFound',
+    InvalidSourceFiles = 'InvalidSourceFiles',
+    CreateUploadUrlError = 'CreateUploadUrlError',
+    UploadArtifactToS3Error = 'UploadArtifactToS3Error',
+    DefaultError = 'DefaultError',
+    InvalidSourceZip = 'InvalidSourceZip',
+    CreateCodeScanError = 'CreateCodeScanError',
+    CreateCodeScanFailedError = 'CreateCodeScanFailedError',
+    SecurityScanTimedOutError = 'SecurityScanTimedOutError',
+    CodeScanJobFailedError = 'CodeScanJobFailedError',
+}
+
+export const mapErrorToCustomerFacingMessage: Record<ErrorCodes, string> = {
```

Comment:
why not just store the "customer facing message" on the error? there is way too much over-abstraction and indirection in this PR